vertx-pojo-mapper-common/src/main/java/de/braintags/io/vertx/pojomapper/dataaccess/query/impl/Query.java

```diff
@@ -480,4 +461,26 @@ public ISearchConditionContainer and(ISearchCondition... searchConditions) {
   public ISearchConditionContainer or(ISearchCondition... searchConditions) {
     return new QueryOr(searchConditions);
   }
+
+  /**
+   * Create a new field condition object with the given values. Checks if the value is a variable. If yes, creates a
+   * {@link VariableFieldCondition} to replace the variable with its actual value during execution.
+   *
+   * @param field
+   *          the field of the condition
+   * @param operator
+   *          the logic operator for the condition
+   * @param value
+   *          the value of the condition
+   * @return a new field condition object
+   */
+  private IFieldCondition createFieldCondition(String field, QueryOperator operator, Object value) {
+    if (value instanceof String && StringUtils.isNotBlank((String) value)) {
+      String stringValue = (String) value;
+      if (stringValue.startsWith("${") && stringValue.endsWith("}")) {
```

Comment:
Well, that would be incorrect syntax. I could trim the value, but I don't know if there should really be much more operations during a string check, since most non-variable values will also be string, so we would trim unnecessarily most of the time.